tools/DeviceExplorer/DeviceExplorer/MainForm.cs

```diff
@@ -403,9 +403,11 @@ private async void UpdateListOfDevices()
         {
             try
             {
+                listDevicesButton.Enabled = false;
                 await updateDevicesGridView();
                 devicesListed = true;
                 listDevicesButton.Text = "Refresh";
+                listDevicesButton.Enabled = true;
```

Comment:
>listDevicesButton.Enabled = true; [](start = 16, length = 33)

Move out from the try-catch into the finally otherwise in a case of exception the button will never be enabled